src/overlays/actors/ovl_Bg_Danpei_Movebg/z_bg_danpei_movebg.c

```diff
@@ -27,38 +36,181 @@ ActorInit Bg_Danpei_Movebg_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80AF753C[] = {
+static u16 D_80AF7530[] = { OBJECT_D_LIFT, OBJECT_DANPEI_OBJECT };
+
+static Gfx* D_80AF7534[] = { gDampeGraveBrownElevatorDL };
+
+static CollisionHeader* D_80AF7538[] = { &gDampeGraveBrownElevatorCol };
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneScale, 1500, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 1100, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_80AF753C[];
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/func_80AF6DE0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/BgDanpeiMovebg_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/func_80AF6EA8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/BgDanpeiMovebg_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/BgDanpeiMovebg_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/func_80AF705C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/func_80AF70FC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/func_80AF71FC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/func_80AF72F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/func_80AF7354.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/func_80AF746C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Danpei_Movebg/func_80AF74CC.s")
+s32 func_80AF6DE0(PlayState* this, ActorPathing* actorPathing) {
+    Actor* actor;
+    s32 res;
+
+    actor = actorPathing->actor;
+    res = 0;
+    if (actorPathing->distSqToCurPoint < actor->speed * actor->speed) {
+        res = 1;
+    } else {
+        actor->world.rot.x = actorPathing->rotToCurPoint.x;
+        actor->world.rot.y = actorPathing->rotToCurPoint.y;
+    }
+    return res;
+}
+
+void BgDanpeiMovebg_Init(Actor* thisx, PlayState* play) {
+    BgDanpeiMovebg* this = THIS;
+    DynaPolyActor_Init(&this->dyna, 1);
+    this->bankIdx = SubS_GetObjectIndex(D_80AF7530[BGDANPEIMOVEBG_GET_TYPE(thisx)], play);
```

Comment:
```suggestion
    this->objectSlot = SubS_GetObjectSlot(D_80AF7530[BGDANPEIMOVEBG_GET_TYPE(thisx)], play);
```
This was renamed very recently